app/lib/cognito.server.ts

```diff
@@ -58,6 +69,55 @@ export function extractAttributeRequired(
   return value
 }
 
+export async function getCognitoUserFromSub(sub: string) {
+  const escapedSub = sub.replaceAll('"', '\\"')
+  const user = (
+    await cognito.send(
+      new ListUsersCommand({
+        UserPoolId,
+        Filter: `sub = "${escapedSub}"`,
+      })
+    )
+  )?.Users?.[0]
+
+  if (!user?.Username)
+    throw new Response('Requested user does not exist', {
+      status: 400,
+    })
+
+  const groups = (await listGroupsForUser(user.Username)).map(
+    (group) => group.GroupName
+  )
+  return { ...user, groups }
```

Comment:
Seems like this is mixing up two distinct pieces of functionality in one function.

Reply:
Yea, good catch. I will fix this in the next commit